asyncscope.md

```diff
@@ -2317,7 +2317,225 @@ spec here
+
+[3]{.pnum} Any call to `t.try_associate()` for a `token` object `t`
+    referring to a `simple_counting_scope` object `s` which are
+    sequenced after a call to `s.close()` return a disengaged
+    association ([exec.scounting.assoc]{.sref}, p1).
+
+`auto join() noexcept;`
+
+[4]{.pnum} _Returns:_ `@_make_sender_@(@_join-t_@, this)`
+
+[5]{.pnum} The exposition-only class template `@_impls-for_@`
+    ([exec.snd.general]{.sref}) is specialized for `@_join-t_@`
+    as follows:
+
+```
+    template <>
+    struct @_impls-for_@<@_join-t_@>: @_default-impls_@> {
+        template <class Receiver>
+        struct @_state_@ {  // @_exposition-only_@
+            simple_counting_scope* @_scope_@; // @_exposition-only_@
+            remove_cvref_t<Receiver>& @_receiver_@; // @_exposition-only_@
+
+            void @_complete_@() { // @_exposition-only_@
+                set_value(std::move(@_receiver_@));
+            }
+        };
+
+        static constexpr auto @_get-state_@ =
+          []<class Receiver>(auto&& sender, Receiver& receiver) {
+            auto[_, self] = sender;
+            return @_state_@<Receiver>(self, receiver);
+          };
+        static constexpr auto @_start_@ =
+            [](auto& s, auto&) { @_see-below_@; };
+    };
+```
+
+[7]{.pnum} In the function object used to initialize
+    `@_impls-for_@<@_join-t_@>::@_start_@` let `state` be `s.@_scope_@->@_state_@`.
+    If `state` is
+
+- [7.1]{.pnum} `@_unused_@`, `@_unused-and-closed_@`, or `@_joined_@`,
+    `s.@_complete_@()` is invoked;
+- [7.2]{.pnum} `@_open_@`, changes the state of `*s.@_scope_@` to `@_open-and-joining_@`;
+- [7.3]{.pnum} `@_closed_@`, changes the state of `*s.@_scope_@` to `@_closed-and-joining_@`;
+
+[8]{.pnum} If `s.@_complete_@()` was not invoked, registers `s` with
+    `*s.@_scope_@` to have `s.@_complete_@()` invoked when `s.@_scope_@->@_count_@` becomes
+    zero.
+
+##### Token [exec.scounting.token]
+
+`template <sender Sender>`
+`Sender&& wrap(Sender&& snd) const noexcept`;
+
+[1]{.pnum} _Returns:_ `std::forward<Sender>(snd);`
+
+`assoc try_associate() const;`
+
+[2]{.pnum} _Effects:_ **TODO**
+
+[3]{.pnum} _Returns:_ **TODO**
+
+##### Assoc [exec.scounting.assoc]
+
+[1]{.pnum} An object `a` of type `assoc` is _disengaged_ if `a.@_scope_@
+    == nullptr` is `true` and _engaged_ otherwise.
+
+`assoc(const assoc& other) noexcept;`
+
+[2]{.pnum} _Effects_: If `other` is engaged evalutes
+    `*this = other.@_scope_@->get_token().try_associate()`.
+
+`assoc(assoc&& other) noexcept;`
+
+[3]{.pnum} _Effects_: Initializes `@_scope_@` with
+    `exchange(other.@_scope_@, nullptr)`.
+
+`~assoc();`
+
+[4]{.pnum} _Effects_: If the object is disengaged, does nothing;
+    otherwise decrements `@_scope_@->@_count_@`. If `@_scope_@->@_count_@`
+    is zero after decrementing, changes the state of `*@_scope_@`
+    to `@_joined_@` and calls `@_complete_@()` on all objects
+    registered with `*@_scope_@`.
```

Comment:
We only transition to _`joined`_ from _`open-and-joining`_ and _`closed-and-joining`_.

```suggestion
    otherwise decrements `@_scope_@->@_count_@`. If `@_scope_@->@_count_@`
    is zero after decrementing and `@_scope_@->@_state_@` is `@_open-and-joining_@`
    or `@_closed-and-joining_@`, changes the state of `*@_scope_@` to `@_joined_@`
    and calls `@_complete_@()` on all objects registered with `*@_scope_@`.
```

Reply:
OK. That should be fine. I should also have a look at the state diagram.